### PR TITLE
Refactor/consistent action dispatch

### PR DESCRIPTION
**Pull request by tkah**
* **Problem:**
   1. Redux actions are not dispatched consistently. 
   1. Component database calls would be better served by already existing actions.
* **Solution:**
   1. Use shorthand pre-bound action creators as outlined [here](http://blog.isquaredsoftware.com/2016/10/idiomatic-redux-why-use-action-creators/).
   1. Call actions over separate database calls. (Really only one in `coinstac-ui/app/render/components/projects/form-project-controller.js`)
* **Tests:**
   1. `npm run test` in base directory
   1. `npm run test:e2e` in coinstac-ui package directory.

### CODE REVIEW

File: packages/coinstac-ui/app/render/components/user-account-controller.js [hunk 1]
```diff
@@ -38,7 +37,7 @@ class UserAccountController extends Component {
 
   render() {
     return (
-      <UserAccount logout={this.logout} {...this.props} />
```
Comment (swashcap) [inline packages/coinstac-ui/app/render/components/user-account-controller.js]: Why change this property’s name?

File: packages/coinstac-ui/app/render/components/dashboard-consortia.js [hunk 2]
```diff
@@ -23,12 +23,12 @@ class DashboardConsortia extends Component {
 
           return (
             <ConsortiumCard
-              deleteConsortium={() => dispatch(deleteConsortium(id))}
```
Comment (swashcap) [inline packages/coinstac-ui/app/render/components/dashboard-consortia.js]: In cases where you’re using multiple properties/functions from `this.props` it’s nice to destructure:

```js
const {
  deleteConsortium,
  joinConsortium,
  leaveConsortium,
} = this.props;

// ...
 <ConsortiumCard
  deleteConsortium={() => deleteConsortium(id)}
  joinConsortium={() => joinConsortium(id, username)}
  leaveConsortium={() => leaveConsortium(id, username)}
/>
```

A little nicer to read.

File: packages/coinstac-ui/app/render/components/consortium/consortium-controller.js [hunk 3]
```diff
@@ -151,15 +151,23 @@ ConsortiumController.contextTypes = {
 ConsortiumController.displayName = 'ConsortiumController';
 
 ConsortiumController.propTypes = {
+  addConsortiumComputationListener: PropTypes.func,
```
Comment (swashcap) [inline packages/coinstac-ui/app/render/components/consortium/consortium-controller.js]: Holy additional prop types. 😲

**Review by swashcap — CHANGES_REQUESTED** [review]

File: packages/coinstac-ui/app/render/components/consortium/consortium-controller.js (hunk 3 above)
Comment (tkah) [reply packages/coinstac-ui/app/render/components/consortium/consortium-controller.js]: Yeah, using this manner of binding means more prop types unfortunately.

File: packages/coinstac-ui/app/render/components/dashboard-consortia.js (hunk 2 above)
Comment (tkah) [reply packages/coinstac-ui/app/render/components/dashboard-consortia.js]: 👍

File: packages/coinstac-ui/app/render/components/user-account-controller.js (hunk 1 above)
Comment (tkah) [reply packages/coinstac-ui/app/render/components/user-account-controller.js]: Adding a new prop called `logout` to take the place of the previously dispatched action `logout` meant that when we pass into `UserAccount` a function named `logout` referencing AccountController's local function in addition to using the spread operator to pass in all the props, including the new `logout` prop, we see issues. The UserAccount component would have two props named `logout`. So, I just renamed the local function.

**Comment by swashcap** [comment]
@tkah Want to resolve the conflicts? Then I can test and merge.

File: packages/coinstac-ui/app/render/components/user-account-controller.js (hunk 1 above)
Comment (swashcap) [reply packages/coinstac-ui/app/render/components/user-account-controller.js]: Looks like `UserAccount` only needs a `user` prop, but that makes total sense. We might want to pass more stuff from `UserAccount Controller` to `UserAccount` in the future.

File: packages/coinstac-ui/app/render/components/consortium/consortium-controller.js (hunk 3 above)
Comment (swashcap) [reply packages/coinstac-ui/app/render/components/consortium/consortium-controller.js]: Comes with the territory I guess :+1:

**Review by swashcap — APPROVED** [review]

File: packages/coinstac-ui/app/render/components/dashboard-home.jsx [hunk 4]
```diff
@@ -21,33 +20,38 @@ class DashboardHome extends Component {
...
   componentWillUpdate() {
     const {
       computations,
       consortia,
-      dispatch,
+      fetchRemoteResultsForUser,
       username,
     } = this.props;
 
-    console.log('update');
```
Comment (swashcap) [inline packages/coinstac-ui/app/render/components/dashboard-home.jsx:41]: Whoa, nice catch.

**Pull request merged**